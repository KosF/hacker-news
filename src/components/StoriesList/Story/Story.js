import React from "react";
import PropTypes from "prop-types";
import TimeAgo from "timeago-react";

import {
  ArticleTitle,
  ArticleTitleLink,
  ArticleSubTitle,
} from "./Story.styles";

const Story = ({ data }) => (
  <article className="border-bottom p-3">
    <header className="mb-2">
      <ArticleTitle>
        {data.url ? (
          <ArticleTitleLink href={data.url} target="_blank">
            {data.title}
          </ArticleTitleLink>
        ) : (
          data.title
        )}
      </ArticleTitle>
    </header>

    <ArticleSubTitle>
      by {data.author.id} ({data.author.karma}){", "}
      <TimeAgo datetime={new Date(data.time * 1000).toString()} /> | score:{" "}
      {data.score}
    </ArticleSubTitle>
  </article>
);

Story.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string,
    time: PropTypes.number,
    score: PropTypes.number,
    author: PropTypes.shape({
      id: PropTypes.string.isRequired,
      karma: PropTypes.number,
    }),
  }).isRequired,
};

export default Story;
