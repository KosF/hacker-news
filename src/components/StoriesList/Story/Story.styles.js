import styled from "styled-components";

export const ArticleSubTitle = styled.p`
  font-size: 12px;
  margin: 0;
`;

export const ArticleTitle = styled.h1`
  color: #f60;
  margin: 0;
  font-size: 18px;
`;

export const ArticleTitleLink = styled.a`
  color: inherit;

  &:hover {
    color: inherit;
  }
`;
