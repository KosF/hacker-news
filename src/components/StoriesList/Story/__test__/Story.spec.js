import React from "react";

import Story from "../Story";

describe("<Story />", () => {
  const props = {
    data: {
      title: "Title",
      time: 1,
      score: 1,
      author: {
        id: "1",
        karma: 1,
      },
    },
  };

  const wrapper = global.mount(<Story {...props} />);

  it("should render correct article", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should add a title link when the 'url' prop is exists", () => {
    expect(
      wrapper.find("header h1").childAt(0).childAt(0).isEmpty()
    ).toBeTruthy();
    wrapper.setProps({
      data: {
        ...props.data,
        url: "url",
      },
    });
    expect(wrapper.find("header h1").childAt(0).childAt(0).type()).toEqual("a");
  });
});
