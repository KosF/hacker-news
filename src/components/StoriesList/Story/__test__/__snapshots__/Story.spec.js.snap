// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Story /> should render correct article 1`] = `
<Story
  data={
    Object {
      "author": Object {
        "id": "1",
        "karma": 1,
      },
      "score": 1,
      "time": 1,
      "title": "Title",
    }
  }
>
  <article
    className="border-bottom p-3"
  >
    <header
      className="mb-1"
    >
      <styled.h1>
        <h1
          className="sc-AxirZ jiZUVp"
        >
          Title
        </h1>
      </styled.h1>
    </header>
    <styled.p>
      <p
        className="sc-AxjAm fKuqGd"
      >
        by 
        1
         (
        1
        )
        , 
        <TimeAgo
          className=""
          datetime="Thu Jan 01 1970 03:00:01 GMT+0300 (Eastern European Standard Time)"
          live={true}
        >
          <time
            className=""
          >
            50 years ago
          </time>
        </TimeAgo>
         | score:
         
        1
      </p>
    </styled.p>
  </article>
</Story>
`;
